### PR TITLE
standard use snake case for meta request

### PR DESCRIPTION
**Pull request by ayocodingit**
# Overview

- standard use snake case for meta
- standard use snake case for request

cc: @jabardigitalservice/jds-backend 

## Evidence
- title: standard use snake case for meta request
- project: Desa Digital
- participants: @ayocodingit @rizkyrmsyah @tukangremot @sandisunandar99 @rachadiannovansyah @khihadysucahyo

### CODE REVIEW

**Review by AldiNFitrah — APPROVED** [review]

**Review by rizkyrmsyah — APPROVED** [review]

File: src/modules/partners/partner_service.ts [hunk 1]
```diff
@@ -5,15 +5,20 @@ export namespace Partner {
...
+      per_page: requestQuery.per_page,
+      current_page: requestQuery.current_page
     })
 
     const result: Entity.ResponseFindAll = {
       data: items.data,
       meta: {
-        lastUpdate: items.data.length > 0 ? items.data[0].created_at : null,
-        ...items.pagination
+        last_update: items.data.length > 0 ? items.data[0].created_at : null,
```
Comment (khihadysucahyo) [inline src/modules/partners/partner_service.ts:15]: I think `items.data.length` is simply enough, since you're only returning a boolean

Comment (ayocodingit) [reply src/modules/partners/partner_service.ts]: thanks @khihadysucahyo :smile:

**Review by tukangremot — APPROVED** [review]

File: src/helpers/paginate.ts [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 }) => {
-  const perPageNumber = Number(requestQuery.perPage)
-  const currentPageNumber = Number(requestQuery.currentPage)
+  const perPageNumber = Number(requestQuery.per_page)
+  const currentPageNumber = Number(requestQuery.current_page)
 
-  const perPage = requestQuery.perPage && !isNaN(perPageNumber) ? perPageNumber : 20
-  const currentPage = requestQuery.currentPage && !isNaN(currentPageNumber) ? currentPageNumber : 1
+  const perPage = requestQuery.per_page && !isNaN(perPageNumber) ? perPageNumber : 20
+  const currentPage = requestQuery.current_page && !isNaN(currentPageNumber) ? currentPageNumber : 1
```
Comment (khihadysucahyo) [inline src/helpers/paginate.ts:9]: is it necessary to check `request Query.current page`? if you have put in to currentPage variable.

**Review by hafizulf — APPROVED** [review]

File: src/helpers/paginate.ts (hunk 2 above)
Comment (ayocodingit) [reply src/helpers/paginate.ts:9]: yes, tested on local success

Comment (khihadysucahyo) [reply src/helpers/paginate.ts:9]: hmm okay
i just wondering maybe u just need to check only `currentPageNumber`, so its just:
`const currentPage = !isNaN(currentPageNumber) ? currentPageNumber : 1`

Comment (ayocodingit) [reply src/helpers/paginate.ts:9]: yes, that's why we added check `isNaN`

**Review by rachadiannovansyah — APPROVED** [review]

File: src/modules/partners/partner_service.ts [hunk 3]
```diff
@@ -5,15 +5,20 @@ export namespace Partner {
...
       meta: {
-        lastUpdate: items.data.length > 0 ? items.data[0].created_at : null,
-        ...items.pagination
+        last_update: items.data.length ? items.data[0].created_at : null,
+        current_page: items.pagination.currentPage,
+        from: items.pagination.from,
+        last_page: items.pagination.lastPage || 0,
+        per_page: items.pagination.perPage,
+        to: items.pagination.to,
+        total: items.pagination.total || 0,
```
Comment (khihadysucahyo) [inline src/modules/partners/partner_service.ts:21]: penasaran doangnih sama typescript..
i dont know what its return of `items.pagination.total` , if struct always return 0 for number data type.
maybe u don't  need to use it ` || 0`

Comment (khihadysucahyo) [reply src/modules/partners/partner_service.ts:21]: hmm okay, kirain bakal selalu default return 0 sih jika false/null, kalo data typenya number (kaya di go)

File: src/helpers/paginate.ts (hunk 2 above)
Comment (ayocodingit) [reply src/helpers/paginate.ts]: updated @khihadysucahyo  e785107

**Comment by khihadysucahyo** [comment]
daddy's service 😘

**Review by khihadysucahyo — APPROVED** [review]
daddy's service 😘

**Pull request merged**